design/noobaa/system-delete.md

```diff
@@ -0,0 +1,10 @@
+♜ [Rook NooBaa Design](README.md) /
+# System Delete
+
+The operator will detect deletion of a system CR and will folloup by deleting all the owned resources.
+
+In general we can leave this to Garbage Collection as described here:
+
+https://kubernetes.io/docs/concepts/workloads/controllers/garbage-collection/
+
+While this works, and there are no known dependencies on the order to deletions, we will prefer to call it explicitly and leave the GC to edge cases.
```

Comment:
The rook-ceph operator handles the cluster CR deletion with two approaches:
- A `finalizer` is added to the CephCluster CR after it is created. This means that the CR is only marked for a deletion until the operator runs some cleanup code. Then the operator removes the finalizer, allowing the CR to actually be deleted. All code that needs to perform cleanup of resources should be run here.
   - Code for [adding and removing finalizers](https://github.com/rook/rook/blob/master/pkg/operator/ceph/cluster/controller.go#L676)
   - Code for [handling the cleanup](https://github.com/rook/rook/blob/master/pkg/operator/ceph/cluster/controller.go#L590)
   - Finalizers should be used sparingly since they will block resource cleanup if the operator isn't able to handle the deletion request.
- The OwnerReferences field is set to the CephCluster CR for each K8s resource that is created. This way, all of the deployments, secrets, configmaps, and other resources created by the operator are automatically removed by K8s when the CR is deleted.
   - See the [SetOwnerRefs()](https://github.com/rook/rook/blob/master/pkg/operator/k8sutil/resources.go#L77) method

Reply:
Thanks for the detailed explanation! Very useful.
I included a finalizer in the design for deletion of Backing Stores.
And will set the OwnerReference for all the subresources created.
Resolving.